File: server/src/Models/DMToolkitModels/Items/Bases/ItemInstanceBase.cs
using DMToolkit.API.Models.DMToolkitModels.Instances;

namespace DMToolkit.API.Models.DMToolkitModels.Items.Bases;

public abstract class ItemInstanceBase
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string ItemType { get; set; } = "Item";
    public int Quantity { get; set; } = 1;
    public bool IsEquipped { get; set; } = false;
    public string DefinitionId { get; set; } = string.Empty;
    public ItemDefinitionBase? Definition { get; set; } = null;
    public string CharacterId { get; set; } = string.Empty;
    public Character? Character { get; set; } = null;
}

File: flutter-client/dm_toolkit/lib/models/dm_toolkit/items/bases/item_instance_base.dart
import 'package:dm_toolkit/models/dm_toolkit/items/bases/item_definition_base.dart';

abstract class ItemInstanceBase {
  String id;
  String itemType;
  int quantity;
  bool isEquipped;
  ItemDefinitionBase definition;

  ItemInstanceBase({
    this.id = '',
    this.itemType = 'Item',
    this.quantity = 1,
    this.isEquipped = false,
    required this.definition
  });
}